libraries/tools/kotlin-gradle-plugin/src/main/kotlin/org/jetbrains/kotlin/gradle/targets/native/spm/SwiftPackageExtension.kt

```diff
@@ -0,0 +1,380 @@
+import org.gradle.api.tasks.Input
+import org.gradle.api.tasks.Nested
+import org.gradle.api.tasks.Optional
+import org.gradle.util.ConfigureUtil
+
+class SwiftPackageExtension(private val project: Project) {
+    @Input
+    lateinit var name: String
+
+    // TODO: List<String> or String
+    @Input
+    @Optional
+    var defaultLocalization: List<String>? = null
+
+    @Nested
+    @Optional
+    var platforms: List<SupportedPlatform>? = null
+
+    @Nested
+    var productsLibraries: List<Products.ProductType.Library> = listOf()
+
+    @Nested
+    var productsExecutables: List<Products.ProductType.Executable> = listOf()
+
+    @Nested
+    var dependencies: List<Dependencies.Package> = listOf()
+
+    @Nested
+    var targetsRegular: List<Targets.Target> = listOf()
+
+    @Nested
+    var targetsExecutable: List<Targets.Target> = listOf()
+
+    @Nested
+    var targetsTest: List<Targets.Target> = listOf()
+
+    // TODO: swiftLanguageVersions: [SwiftVersion]? = nil,
+    // TODO: cLanguageStandard: CLanguageStandard? = nil,
+    // TODO: cxxLanguageStandard: CXXLanguageStandard? = nil
+
+    fun platforms(configure: SupportedPlatform.() -> Unit) {
+        val supportedPlatform = SupportedPlatform().apply(configure)
+        platforms = supportedPlatform.platforms
+    }
+
+    fun products(configure: Products.() -> Unit) {
+        val product = Products().apply(configure)
+
+        productsLibraries = product.libraries
+        productsExecutables = product.executables
+    }
+
+    fun dependencies(configure: Dependencies.() -> Unit) {
+        val dependency = Dependencies().apply(configure)
+        dependencies = dependency.dependencies
+    }
+
+    fun targets(configure: Targets.() -> Unit) {
+        val target = Targets().apply(configure)
+
+        targetsRegular = target.regulars
+        targetsExecutable = target.executables
+        targetsTest = target.tests
+    }
+
+    // TODO: add fun with another params (supportedPlatform)
+    class SupportedPlatform() {
+        var platforms = mutableListOf<SupportedPlatform>()
+
+        lateinit var type: PlatformType
+        lateinit var version: String
+
+        constructor(type: PlatformType, version: String) : this() {
+            this.type = type
+            this.version = version
+        }
+
+        fun macOS(version: String) {
+            val platform = SupportedPlatform(PlatformType.macOS, version)
+            platforms.add(platform)
+        }
+
+        fun iOS(version: String) {
+            val platform = SupportedPlatform(PlatformType.iOS, version)
+            platforms.add(platform)
+        }
+
+        fun tvOS(version: String) {
+            val platform = SupportedPlatform(PlatformType.tvOS, version)
+            platforms.add(platform)
+        }
+
+        fun watchOS(version: String) {
+            val platform = SupportedPlatform(PlatformType.watchOS, version)
+            platforms.add(platform)
+        }
+
+        enum class PlatformType {
```

Comment:
MPP plugin contains proper API for platforms, re-use it (see org.jetbrains.kotlin.gradle.plugin.KotlinTarget#getPlatformType)